http-poja/src/main/java/io/micronaut/http/poja/rawhttp/RawHttpBasedServletHttpRequest.java

```diff
@@ -0,0 +1,412 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.poja.rawhttp;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ArgumentConversionContext;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.convert.value.MutableConvertibleMultiValuesMap;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.MutableHttpHeaders;
+import io.micronaut.http.MutableHttpParameters;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.body.ByteBody;
+import io.micronaut.http.codec.MediaTypeCodecRegistry;
+import io.micronaut.http.cookie.Cookie;
+import io.micronaut.http.cookie.Cookies;
+import io.micronaut.http.poja.PojaHttpRequest;
+import io.micronaut.http.poja.util.LimitingInputStream;
+import io.micronaut.http.simple.cookies.SimpleCookies;
+import io.micronaut.servlet.http.body.InputStreamByteBody;
+import rawhttp.cookies.ServerCookieHelper;
+import rawhttp.core.RawHttp;
+import rawhttp.core.RawHttpHeaders;
+import rawhttp.core.RawHttpRequest;
+import rawhttp.core.RawHttpResponse;
+import rawhttp.core.body.BodyReader;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpCookie;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * @author Sahoo.
+ */
+public class RawHttpBasedServletHttpRequest<B> extends PojaHttpRequest<B, RawHttpRequest, RawHttpResponse<Void>> {
+    private final RawHttp rawHttp;
+    private final RawHttpRequest rawHttpRequest;
+    private final ByteBody byteBody;
+    private final RawHttpBasedHeaders headers;
+    private final RawHttpBasedParameters queryParameters;
+
+    public RawHttpBasedServletHttpRequest(
+        InputStream in,
+        ConversionService conversionService,
+        MediaTypeCodecRegistry codecRegistry,
+        ExecutorService ioExecutor,
+        RawHttpBasedServletHttpResponse<Void> response
+    ) {
+        super(conversionService, codecRegistry, (RawHttpBasedServletHttpResponse) response);
+        this.rawHttp = new RawHttp();
+        try {
+            rawHttpRequest = rawHttp.parseRequest(in);
```

Comment:
It seems like Raw HTTP cannot parse requests with `transfer-encoding: chunked`. We will probably need to investigate that.

Reply:
no we shouldn't be investigating and fixing bugs in an immature library. We will have to wait for @wwwsahoo input on replacing this with a mature http parsing solution.

Reply:
You are right. We will need to make sure the new implementation handles this case well, though.